### ISSUE TITLE
Better information when resource returns empty version list

### ISSUE THREAD

**Issue by bleach** [concourse/atc#38]
We have been trying to use the concourse S3 resource to hold some state and
been running into some issues that proved hard to diagnose. We think that when
a resource returns an empty array of versions, atc gets confused and the job is
shown as pending.

We have pipeline containing:
- an S3 resource pointing to a bucket where we keep our terraform state
- a job which runs the terraform command and writes the state to the S3 bucket

```
resources:
- name: s3-bucket
  type: s3
  source:
    access_key_id: {{aws_key}}
    bucket: piotr-state
    region_name: eu-west-1
    secret_access_key: {{secret_key}}
    versioned_file: vpc.tfstate
jobs:
- name: test-s3
  serial: true
  plan:
  - get: s3-bucket
  - task: echo
    config:
      image: docker:///governmentpaas/bosh-init
      inputs:
      - name: s3-bucket
        path: ""
      run:
        path: date
```

This is a cut-down version of our pipeline, which runs on concourse-lite and is
used to bootstrap our real concourse installation.

When we run this for the first time, the S3 bucket contains no files matching 
the versioned_file parameter passed to the S3 resource ie. the resource is
"empty" and has no versions.

- There is no obvious error when clicking on the S3 resource in the UI and the 
job is shown as pending.
- There is no output from the resource visible in the UI, nothing to indicate 
what is going wrong.

We did a bit of debugging and found that the call to check passed in a null
version, as might be expected from reading
http://concourse.ci/implementing-resources.html.

The output from the S3 resource check executable contained an empty array of versions.

When a resource is empty, how should it respond to a check? With an empty array
of versions or a single null version?

If a resource does respond in a way that's unexpected or not allowed by
atc/concourse, we think it would be helpful to make the job show as failed
along with a reason. We spent a fair amount of time thinking that the job
itself was hanging.

**Comment by vito** [comment]
A resource returning an empty versions is valid, and what everything stabilizes on (there won't always be something new). It sounds like this is working as expected; if there's no versioned file in the bucket, there are no versions, and so the job can't run. Showing it as "pending" forever with no indication why doesn't help though.

We have a separate story for showing which resource a job is waiting on: https://www.pivotaltracker.com/story/show/99634350

see also #114

**Comment by bleach** [comment]
Thanks for the response, it looks like our issue is the same as concourse/concourse#114 and the story to show which resource a job is waiting on would have saved us time debugging. 

Is there a way for us to indicate our interest in the feature?

**Comment by vito** [comment]
@bleach leaving a comment should be enough
